src/components/folder/LinkList.js

```diff
@@ -1,66 +1,135 @@
 import kebab from "../../images/kebab.png";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import AddModal from "../modals/AddModal";
+import DeleteModal from "../modals/DeleteModal";
 
-function LinkListItem({ linkData }) {
+function LinkListItem({ folderListData, linkData }) {
   const [kebabOn, setKebabOn] = useState(false);
+  const [addModalOn, setAddModalOn] = useState(false);
+  const [deleteModalOn, setDeleteModalOn] = useState(false);
   const url = linkData.url;
-  let image = linkData.image_source;
-  if (!image) {
-    image = card;
-  }
-  const handleKebabIn = (e) => {
+  let image = linkData.image_source || card;
+
+  const handleKebab = (e, isIn) => {
     e.preventDefault();
-    setKebabOn(true);
-  };
-  const handleKebabOut = () => {
-    setKebabOn(false);
+    setKebabOn(isIn);
   };
+
+  function handleAddModalOn() {
+    setAddModalOn(true);
+  }
+  function handleAddModalOff() {
+    setAddModalOn(false);
+  }
+  function handleDeleteModalOn() {
+    setDeleteModalOn(true);
+  }
+  function handleDeleteModalOff() {
+    setDeleteModalOn(false);
+  }
+
+  function kebabMenuBlur(e) {
+    const kebabMenuButtons = document.querySelectorAll(".kebab-menu");
+
+    if (
+      e.relatedTarget !== kebabMenuButtons[0] &&
+      e.relatedTarget !== kebabMenuButtons[1]
+    ) {
+      handleKebab(e, false);
+    }
+  }
   return (
-    <a
-      href={url}
-      className="card flex flex-col"
-      target="_blank"
-      rel="noopener noreferrer"
-    >
-      <div
-        key={image}
-        className="card-image"
-        style={{ backgroundImage: `url(${image})` }}
+    <>
+      <a
+        href={url}
+        className="card flex flex-col"
+        target="_blank"
+        rel="noopener noreferrer"
       >
-        <button className="star">
-          <img src={star} />
-        </button>
-      </div>
-      <div className="card-content">
-        <div onClick={handleKebabIn} onBlur={handleKebabOut}>
-          <button className="kebab">
-            <img src={kebab} />
+        <div
+          key={image}
+          className="card-image"
+          style={{ backgroundImage: `url(${image})` }}
+        >
+          <button className="star">
+            <img src={star} />
           </button>
-          {kebabOn ? (
-            <div className="kebab-menus">
-              <button className="kebab-menu">삭제하기</button>
-              <button className="kebab-menu">폴더에 추가</button>
-            </div>
-          ) : (
-            <></>
-          )}
         </div>
-        <p>{getDaysAgo(linkData.created_at)}</p>
-        <h2>{linkData.title}</h2>
-        <p>{formatDate(linkData.created_at)}</p>
-      </div>
-    </a>
+        <div className="card-content">
+          <div>
+            <button
+              onClick={(e) => {
+                handleKebab(e, true);
+              }}
+              onBlur={(e) => {
+                kebabMenuBlur(e);
+              }}
```

Comment:
사소하지만 이런 경우에, e는 이벤트라 그대로 전달되어서, 이렇게만 적으셔도 원하시는 대로 동작합니다~


```suggestion
              onBlur={kebabMenuBlur}
```

아래 글에서 Without Parameter 모습 보시면 됩니다!
https://stackoverflow.com/questions/42597602/react-onclick-pass-event-with-parameter